src/components/worker/worker.py

```diff
@@ -507,22 +519,38 @@ async def get_pcd(self) -> CapturedData:
                 "Error getting PCD: depth frame queue not configured for current OAK camera."
             )
 
-        msg = None
+        msg: Optional[dai.MessageGroup] = None
         while msg is None:
             try:
                 msg = self.pc_queue.tryGet()
             except Empty:
-                self.logger.debug("Couldn't get color frame: frame queue is empty.")
+                self.logger.debug("Couldn't get point cloud: queue is empty.")
             await asyncio.sleep(0.01)
 
+        message_names = msg.getMessageNames()
+        if not ("rgb" in message_names and "pcl" in message_names):
+            points = msg["pcl"].getPoints().astype(np.float64)
+            timestamp = msg["pcl"].getTimestamp().total_seconds()
+            downsampled_points = self._downsample_pcd(
+                points, MAX_GRPC_MESSAGE_BYTE_COUNT
+            )
```

Comment:
Is this the defensive code? To not try to make colour or depth point clouds if the sensor has no pcl capabilities?